Automattic/jetpack
# RNA: IDC migration

#### Changes proposed in this Pull Request:
* introduce the package-wide state
* add a package-wide global state variable called `isActionInProgress`, meaning that there's something going on and no buttons should be clickable
* introduce the IDC "migrate" functionality, send the API request to set the "migrate" flag
* after the flag is set, display the "migrated" screen

#### Jetpack product discussion
p9dueE-3kA-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Run the following DB query to remove the IDC migration flag:
```
DELETE FROM `wp_options` WHERE option_name = 'jetpack_migrate_for_idc';
```
2. Go to "Tools -> Connection Manager".
3. The "Safe Mode" screen should appear. Click "Move your settings", all other buttons should get greyed out immediately. The "Move your settings" label should get replaced with a loader.
4. The button should display the loader animation, then you see the "Migrated Successfully" screen. Confirm that it looks like the screenshot. Adjust the screen size and make sure it looks good on smaller screens too.
5. Click "Got it, thanks". The page should reload.
6. Run the following DB query and confirm that the flag was set:
```
SELECT * FROM `wp_options` WHERE option_name = 'jetpack_migrate_for_idc';
```
7. After the page has reloaded, click "safe mode", confirm that all the buttons got greyed out and became non-clickable.
8. The page should reload again, click "Create a fresh connection". The button's label should get replaced with a loder, all other buttons should get greyed out and non-clickable for a few seconds, then get back to normal.

#### The screenshot
<img width="1146" alt="Screen Shot 2021-10-08 at 2 40 06 PM" src="https://user-images.githubusercontent.com/1341249/136609838-6f3a0bdb-d9c2-467b-a05a-cb09d1567165.png">
